src/sockets.js

```diff
@@ -1,67 +1,169 @@
+const logger = require('winston')
+const Encryption = require('../utils/encryptionStore')
+const LightningServices = require('../utils/lightningServices')
 
 module.exports = (
   /** @type {import('socket.io').Server} */
   io
 ) => {
-  const Mediator = require("../services/gunDB/Mediator/index.js");
-  
+  const Mediator = require('../services/gunDB/Mediator/index.js')
+
+  // This should be used for encrypting and emitting your data
+  const emitEncryptedEvent = ({ eventName, data, socket }) => {
+    try {
+      if (Encryption.isNonEncrypted(eventName)) {
+        return socket.emit(eventName, data)
+      }
+
+      const deviceId = socket.handshake.query['x-shockwallet-device-id']
+      const authorized = Encryption.isAuthorizedDevice({ deviceId })
+
+      if (!deviceId) {
+        throw {
+          field: 'deviceId',
+          message: 'Please specify a device ID'
+        }
+      }
+
+      if (!authorized) {
+        throw {
+          field: 'deviceId',
+          message: 'Please exchange keys with the API before using the socket'
+        }
+      }
+
+      const encryptedMessage = Encryption.encryptMessage({
+        message: data,
+        deviceId
+      })
+
+      return socket.emit(eventName, encryptedMessage)
+    } catch (err) {
+      logger.error(
+        `[SOCKET] An error has occurred while encrypting an event (${eventName}):`,
+        err
+      )
+
+      return socket.emit('encryption:error', err)
+    }
+  }
+
+  const parseJSON = data => {
+    try {
+      if (typeof data === 'string') {
+        return JSON.parse(data)
+      }
+
+      return data
+    } catch (err) {
+      return data
+    }
+  }
+
+  const decryptEvent = ({ eventName, data, socket }) => {
+    try {
+      const deviceId = socket.handshake.query['x-shockwallet-device-id']
+      if (Encryption.isNonEncrypted(eventName)) {
+        return data
+      }
+
+      if (!data) {
+        return data
+      }
+
+      const parsedData = parseJSON(data)
+
+      if (!deviceId) {
+        throw {
+          field: 'deviceId',
+          message: 'Please specify a device ID'
+        }
+      }
+
+      if (!Encryption.isAuthorizedDevice({ deviceId })) {
+        throw {
+          field: 'deviceId',
+          message: 'Please exchange keys with the API before using the socket'
+        }
+      }
+
+      const decryptedKey = Encryption.decryptKey({
+        deviceId,
+        message: parsedData.encryptedKey
+      })
+      const decryptedMessage = Encryption.decryptMessage({
+        message: parsedData.encryptedData,
+        key: decryptedKey,
+        iv: parsedData.iv
+      })
+      const decryptedData = JSON.parse(decryptedMessage)
+      return decryptedData
+    } catch (err) {
+      logger.error(
+        `[SOCKET] An error has occurred while decrypting an event (${eventName}):`,
+        err
+      )
+
+      return socket.emit('encryption:error', err)
+    }
+  }
+
```

Comment:
I'm a bit confused...

Gun Socket was already encrypted, was LND socket unencrypted then?

Reply:
Yeah it wasn't encrypted before and this change should add the encryption functionality for it too